File: node/test/api/arch/member.test.js
/**
 * Copyright (c) 2015-2022, FineX, All Rights Reserved.
 * @author Thomas.Zhuang
 * @date 2022/05/18
 */

import { MemberApiClient } from '../../../src/api/arch/member';
import { API_KEY, HOST } from '../../../src/constants/http_constant';

export const memberTest = 'memberTest';

const RootDeptNo = 1;

const username = 'jianDaoYun';
const name = '小云';

describe('member api test', () => {
    const api = new MemberApiClient(API_KEY, HOST, 'v5');

    test('deptMemberList', async () => {
        const deptMemberList = await api.deptMemberList(RootDeptNo, {
            hasChild: true
        });
        console.log('deptMemberList result:' + JSON.stringify(deptMemberList));
        expect(deptMemberList.users).toBeTruthy();
    });

    test('userCreate', async () => {
        const user = await api.userCreate(name, {
            username,
            departments: [RootDeptNo]
        });
        console.log('userCreate result:' + JSON.stringify(user));
        expect(user.user).toBeTruthy();
        expect(user.user.name).toEqual(name);
        expect(user.user.username).toEqual(username);

    });

    test('userInfo', async () => {
        const user = await api.userInfo(username);
        console.log('userInfo result:' + JSON.stringify(user));
        expect(user.user).toBeTruthy();
        expect(user.user.name).toEqual(name);
        expect(user.user.username).toEqual(username);
    });

    test('userUpdate', async () => {
        const user = await api.userUpdate(username, {
            name: name + '_update'
        });
        console.log('userUpdate result:' + JSON.stringify(user));
        expect(user.user).toBeTruthy();
        expect(user.user.name).toEqual(name + '_update');
    });

    test('userDelete', async () => {
        const response = await api.userDelete(username);
        console.log('userDelete result:' + JSON.stringify(response));
        expect(response.status).toEqual('success');
    });

    test('userImport', async () => {
        const response = await api.userImport([{
            name: name + '_import',
            username: username + '_import',
            departments: [RootDeptNo]
        }]);
        console.log('userImport result:' + JSON.stringify(response));
        expect(response.status).toEqual('success');
    });

    test('userBatchDelete', async () => {
        const response = await api.userBatchDelete([username + '_import']);
        console.log('userBatchDelete result:' + JSON.stringify(response));
        expect(response.status).toEqual('success');
    });
});

File: csharp/test/api/arch/Member.Test.cs
/**
* Copyright (c) 2015-2022, FineX, All Rights Reserved.
* @author Thomas.Zhuang
* @date 2022/06/14
*/
using Src.Api.Jdy;
using Src.Base;

namespace Test;

[Collection("Member")]
[TestCaseOrderer("XUnit.Project.Orderers.AlphabeticalOrderer", "Test")]
public class MemberTest
{
    public static string username = "";
    public static int rootDeptNo = 1;

    private static MemberApiClient api = new MemberApiClient(Global.Host, Global.ApiKey);

    [Fact]
    public async void Test1()
    {
        var option = new Dictionary<string, object>();
        option.Add("hasChild", true);
        var deptMemberList = await api.deptMemberList(rootDeptNo, option);
        Assert.True(deptMemberList != null);
        Console.WriteLine(deptMemberList);
    }

    [Fact]
    public async void Test2()
    {
        var option = new Dictionary<string, object>();
        option.Add("username", "jiandaoyun");
        var user = await api.userCreate("小云", option);
        Assert.True(user != null);
        Assert.Equal("小云", user.Value.GetProperty("user").GetProperty("name").GetString());
        Assert.Equal("jiandaoyun", user.Value.GetProperty("user").GetProperty("username").GetString());
        Console.WriteLine(user);
        username = user.Value.GetProperty("user").GetProperty("username").GetString();
    }

    [Fact]
    public async void Test3()
    {
        var user = await api.userInfo(username);
        Assert.True(user != null);
        Assert.Equal("小云", user.Value.GetProperty("user").GetProperty("name").GetString());
        Assert.Equal("jiandaoyun", user.Value.GetProperty("user").GetProperty("username").GetString());
        Console.WriteLine(user);
    }

    [Fact]
    public async void Test4()
    {
        var option = new Dictionary<string, object>();
        option.Add("name", "小简");
        var user = await api.userUpdate(username, option);
        Assert.True(user != null);
        Assert.Equal("小简", user.Value.GetProperty("user").GetProperty("name").GetString());
        Assert.Equal("jiandaoyun", user.Value.GetProperty("user").GetProperty("username").GetString());
        Console.WriteLine(user);
    }

    [Fact]
    public async void Test5()
    {
        var response = await api.userDelete(username);
        Assert.True(response != null);
        Assert.Equal("success", response.Value.GetProperty("status").GetString());
    }

    [Fact]
    public async void Test6()
    {
        string[] usernames = { username };
        var response = await api.userBatchDelete(usernames);
        Assert.True(response != null);
        Assert.Equal("success", response.Value.GetProperty("status").GetString());
    }

    [Fact]
    public async void Test7()
    {
        Dictionary<string, object>[] users = { };
        var response = await api.userImport(users);
        Assert.True(response != null);
        Assert.Equal("success", response.Value.GetProperty("status").GetString());
    }
}